File: frontend/ProjectManagement.Core/Models/UpdateWorkItemRequest.cs
namespace ProjectManagement.Core.Models;

public sealed record UpdateWorkItemRequest
{
    public required Guid WorkItemId { get; init; }
    public required int ExpectedVersion { get; init; }
    public string? Title { get; init; }
    public string? Description { get; init; }
    public string? Status { get; init; }
    public string? Priority { get; init; }
    public Guid? AssigneeId { get; init; }
    public int? StoryPoints { get; init; }
    public Guid? SprintId { get; init; }
    public int? Position { get; init; }
    
    /// <summary>
    /// Parent work item ID. Set to Guid.Empty to clear the parent.
    /// Leave null to keep the current parent unchanged.
    /// </summary>
    public Guid? ParentId { get; init; }

    /// <summary>
    /// Indicates whether ParentId should be updated (including clearing).
    /// Required because null ParentId could mean "no change" or "clear parent".
    /// </summary>
    public bool UpdateParent { get; init; }
}

File: backend/pm-server/src/api/work_items/update_work_item_request.rs
use serde::Deserialize;

/// Request body for updating a work item
#[derive(Debug, Deserialize)]
pub struct UpdateWorkItemRequest {
    #[serde(default)]
    pub title: Option<String>,
    #[serde(default)]
    pub description: Option<String>,
    #[serde(default)]
    pub status: Option<String>,
    #[serde(default)]
    pub priority: Option<String>,
    #[serde(default)]
    pub assignee_id: Option<String>,
    #[serde(default)]
    pub sprint_id: Option<String>,
    #[serde(default)]
    pub story_points: Option<i32>,
    #[serde(default)]
    pub parent_id: Option<String>,
    /// Set to true to update parent_id (allows clearing parent)
    #[serde(default)]
    pub update_parent: bool,
    #[serde(default)]
    pub position: Option<i32>,
    /// Required: current version for optimistic locking
    pub expected_version: i32,
}
